exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Service.java

```diff
@@ -53,6 +53,24 @@ default void initialize(Fork fork, Configuration configuration) {
     // No configuration
   }
 
+  /**
+   * Performs resuming of previously stopped service instance. This method is called <em>once</em>
+   * after restarting a service instance.
+   *
+   * <p><em>Note that changing the data during resume operations is not allowed
+   * due to violation migration procedure.</em>
+   *
+   * @param configuration the service configuration parameters
+   * @throws ExecutionException if the configuration parameters are not valid (e.g.,
+   *     malformed, or do not meet the preconditions). Exonum will save the error into
+   *     {@linkplain com.exonum.binding.core.blockchain.Blockchain#getCallErrors(long)
+   *     the registry of call errors} if the resuming fails
+   * @see Configurable
+   */
+  default void resume(Configuration configuration) {
```

Comment:
@slowli, @bullet-tooth  In which cases the service needs to be notified of that, if it can't do anything with this configuration, just look at them? I remember that the service instance _objects_ are not supposed to have much state, therefore,

> "As an example, arguments can be used to update the service configuration."

does not seem to make much sense — that configuration must be normally kept in the persistent storage. On top of that, we do *not* provide a similar operation for _node restarts_, when a fresh service instance object is also created.